trio/_tests/test_exports.py

```diff
@@ -118,18 +116,45 @@ def no_underscores(symbols):
         )
         from mypy.api import run
 
-        res = run(["--config-file=", "--follow-imports=silent", str(tmpfile)])
-
-        # clean up created py.typed file
-        if not py_typed_exists:  # pragma: no cover
-            py_typed_path.unlink()
+        mypy_res = run(["--config-file=", "--follow-imports=silent", str(tmpfile)])
 
         # check that there were no errors (exit code 0), otherwise print the errors
-        assert res[2] == 0, res[0]
+        assert mypy_res[2] == 0, mypy_res[0]
         return
+    elif tool == "pyright_verifytypes":
+        import subprocess
+
+        res = subprocess.run(
+            ["pyright", f"--verifytypes={modname}", "--verbose"],
+            capture_output=True,
+        )
+        start_index = res.stdout.find(b"Public modules: ")
+        end_index = res.stdout.find(b"Other referenced symbols: ", start_index)
+        assert start_index != -1, (res.stdout, res.stderr)
+        assert end_index != -1, res.stdout
+        static_names = {
+            x.strip().decode()[len(modname) + 1 :]
+            for x in res.stdout[start_index:end_index].split(b"\n")
+            if modname + "." in x.decode()
+        }
+
+        # don't require verifytypes to see reexported constants from socket
+        if modname == "trio.socket":
+            import socket
+
+            static_names |= set(filter(str.isupper, dir(socket))) & runtime_names
+
+            # unclear why this one is required
+            if sys.platform == "win32" and sys.version_info[:2] == (3, 7):
+                static_names.add("IPPROTO_IPV6")
+
     else:  # pragma: no cover
         assert False
 
+    # remove py.typed file
+    if tool in ("mypy", "pyright_verifytypes") and not py_typed_exists:
+        py_typed_path.unlink()
```

Comment:
Do you know why codecov is saying this line isn't covered by tests when the tests should be doing this?

Reply:
oops, yep. `mypy` has a `return` at the end of its `if`-branch so it never cleans up after it.